### PR TITLE
refactor: add shared ethereum bars

### PR DESCRIPTION
**Pull request by guanbinrui**
## Introduction

Those Ethereum plugins in Maskbook currently share the same buttons for interacting with the Ethereum chain. Add `EthereumWalletConnectedBoundary` and `EthereumERC20TokenApprovedBoundary ` to encapsulate those buttons as shared components across different plugins.

#### `<EthereumWalletConnectedBoundary />`

Renders a `Connect a Wallet` button if no wallet connected and an `Invalid Network` button if non-mainnet was selected. Otherwise, renders its children as a fallback

#### `<EthereumERC20TokenApprovedBoundary />`

Renders two unlocking buttons if the given conditions cannot be reached by the ERC20 token's balance and allowance. Otherwise, renders its children as a fallback.

## Usage

```ts
<EthereumIfWalletConnected>
    <EthereumIfERC20TokenApproved amount={amount} spender={spenderAddress} token={approveToken}>
        <ActionButton onSubmit={onSubmit}>Submit</ActionButton>
    </EthereumIfERC20TokenApproved>
</EthereumIfWalletConnected>
```

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

File: packages/maskbook/src/web3/UI/EthereumIfWalletConnected.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    createStyles({
+        button: {},
+    }),
+)
+
+export interface EthereumIfWalletConnectedProps {
+    children?: React.ReactNode
+}
+
+export function EthereumIfWalletConnected(props: EthereumIfWalletConnectedProps) {
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/UI/EthereumIfWalletConnected.tsx:20]: I have to say this is super strange

Comment (guanbinrui) [reply packages/maskbook/src/web3/UI/EthereumIfWalletConnected.tsx:20]: I added a usage section. Any better name for this?

File: packages/maskbook/src/web3/UI/EthereumIfERC20TokenApproved.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+)
+
+export interface EthereumIfERC20TokenApprovedProps {
+    amount: string
+    spender: string
+    token?: ERC20TokenDetailed
+    children?: React.ReactNode
+}
+
+export function EthereumIfERC20TokenApproved(props: EthereumIfERC20TokenApprovedProps) {
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/UI/EthereumIfERC20TokenApproved.tsx:26]: I suggest to rename it to `EthereumERC20TokenApprovedBoundary` (component form) or make it a hook that returns JSX children. This suggestion also applies to another component

```js
useEnsureEthereumERC20TokenApproved(amount: string, spender: string, token?: T, children?: ReactNode)
```

Comment (guanbinrui) [reply packages/maskbook/src/web3/UI/EthereumIfERC20TokenApproved.tsx]: It's done!

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**